backend/django/core/templates/projects/create/create_wizard_labels.html

```diff
@@ -73,15 +88,59 @@ <h3>Instructions</h3>
 {% endblock %}
 
 {% block scripts_body %}
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-csv/0.71/jquery.csv-0.71.min.js"></script>
 <script type="text/javascript">
-$('.formset_row_label').formset({
-  addText: 'add label',
-  deleteText: 'remove label',
-  prefix: 'label_set',
-  added: function() {
-    // Reset drop down and errors from both columns when adding new rows
-    $('tr.formset_row_label').last().children(':first').find('ul.errorlist li').empty();
+  function handleUpload(event) {
+    if (event.target.files[0] && event.target.files[0].name.includes('.csv')) {
+      $('.overlay').show();
+      $('#loading_text').html("Please wait. Generating labels...");
+      $("#second_progress").addClass("loader");
+      var reader = new FileReader();
+      $('.error-messages').empty();
+      try {
+        reader.readAsText(event.target.files[0]);
+        reader.onload = function(event){
+          var csv = event.target.result;
+          var data = $.csv.toArrays(csv);
+            data.forEach((label, index) => {
+              if(index !== 0) {
+                if($('.formset_row_label').eq(index - 1).length > 0) {
+                  $('.formset_row_label').eq(index - 1).css('display', '');
+                  $('.formset_row_label').eq(index - 1).find('input[type=text]').eq(0).val(label[0]);
```

Comment:
Noticed something new with this. It isn't strictly *wrong*, but we might want to add a check before this line making sure the fields provided are label and description. 

Basically, I found out if you feed this data input one of the sample data files, it takes the first column "Text" as the list of labels with "Label" as the descriptions, thus making one label for each data point.

Thoughts?